.travis.yml

```diff
@@ -1,23 +1,48 @@
 sudo: false
 language: go
 go:
-- 1.5.3
-- 1.6
-before_install:
-- go get github.com/tools/godep
-- if [ ! -d $SNAP_PLUGIN_SOURCE ]; then mkdir -p $HOME/gopath/src/github.com/intelsdi-x; ln -s $TRAVIS_BUILD_DIR $SNAP_PLUGIN_SOURCE; fi # CI for forks not from intelsdi-x
+- 1.5.4
+- 1.6.2
 env:
   global:
-    - SNAP_PLUGIN_SOURCE=/home/travis/gopath/src/github.com/intelsdi-x/snap-plugin-collector-psutil
+    - ORG_PATH=/home/travis/gopath/src/github.com/intelsdi-x
+    - SNAP_PLUGIN_SOURCE=/home/travis/gopath/src/github.com/${TRAVIS_REPO_SLUG}
+    - GO15VENDOREXPERIMENT=1
   matrix:
-    - TEST=unit
+    - TEST=legacy
+    - TEST=build
+matrix:
+  exclude:
+    - go: 1.5.4
+      env: TEST_TYPE=build
+before_install:
+  - go get github.com/tools/godep
+  - if [ ! -d $SNAP_PLUGIN_SOURCE ]; then mkdir -p $HOME/gopath/src/github.com/intelsdi-x; ln -s $TRAVIS_BUILD_DIR $SNAP_PLUGIN_SOURCE; fi # CI for forks not from intelsdi-x
 install:
-- export TMPDIR=$HOME/tmp
-- mkdir -p $TMPDIR
-- cd $SNAP_PLUGIN_SOURCE # change dir into source
-- make deps
+  - export TMPDIR=$HOME/tmp
+  - mkdir -p $TMPDIR
+  - cd $SNAP_PLUGIN_SOURCE # change dir into source
+  - make deps
 script:
-- make check TEST=$TEST 2>&1 # Run test suite
+  - make check TEST=$TEST 2>&1 # Run test suite
+before_deploy:
+  - make all
+  - "./scripts/pre_deploy.sh"
+deploy:
+  provider: s3
+  access_key_id: AKIAINMB43VSSPFZISAA
+  secret_access_key:
+    secure: YvbymtPIqf/VvjBGlZNNTr1bQ5BWvBETEmTq9xprCzfYz5WvNKS7VYDrLmN8ZKF9SJAjpad0a93TCSkR/78Z/awMIKscCWODsYyYfSDkzN+uIqFhFKse6YnaU3795qmt8gZAf4guNQthOYGtlLEwsl3O0vlmesCXAR6Dwi1IGN+t9BUnCgM7OmcTcNpRHoedwIgY0M7c+JbFJtx1vP1+Tc1JL5w8q7PaMdsTPlbhW7w+i+zrjYpuskxcc+A6VoYB+lKf+wrX/uHM8pHSPYJOS5Kp0m8khbLED820geoZ79knUTcxMxrqPsXXk9/QqSK0dR0M5Jey8j2H9jUfAEgN/4nAftSVTJitVnDlIU2blx6Fly/nNkiZHqXw9T87Sdl9ZF/sTOwDJaPnsDbh9gwf9QXeWwi7k+qFBu1IhYtoV6K/ynRVHWBl1SaHOBD3GcaA4rLYQM1Bk2vC4Cs75IA5o1wFsg9OFkPAjwawasaXKypQpP/Zk0hfDS3HsX3fDObHOelH1i5ajqxltZEg4OJCMkKyRfjAHtulxtRTLzxMm+Cj/flBRvr8UvhUEeUuCMDXszatbwKMLLFGY0hgKDqpjCliSnRqnHBfbDhc1t5So02x2ngx4Uw2tzySvWjDCBGRDRNeI+rE78VCD4EQWZCJRH10+Eqdxe209aNLMI/dli0=
+  bucket: snap.ci.snap-telemetry.io
+  region: us-west-2
+  skip_cleanup: true
+  local-dir: build
+  upload-dir: plugin
+  acl: public_read
+  on:
+    repo: intelsdi-x/snap-plugin-collector-psutil
+    branch: build
+    condition: $TEST_TYPE=build && $TRAVIS_GO_VERSION="1.6.2"
```

Comment:
if you'll correct https://github.com/intelsdi-x/snap-plugin-collector-psutil/pull/25/files#r74232025 remember to change this too